### ISSUE TITLE
Feature request: Fully featured 404 "not found" page

### ISSUE THREAD

**Issue by ottok** [CaiJimmy/hugo-theme-stack#936]
### What happened?

Currently is a user follows a rotten link they will be presented with a very plain 404 page. Example from https://optimizedbyotto.com/asdf:

![image](https://github.com/CaiJimmy/hugo-theme-stack/assets/668724/d1a379fc-8e6f-47b4-968d-964612ae8acc)

The page could have some more features/content to help the visitor continue their site journey.

For example, the page could re-use the `search` page and show a search, with the last slug element from the URL as placeholder in the search and results visible. Thus, if the link had for example a small typo, the search would most likely find the result.

Additionally the page could show a sitemap to help users ''scan" with their eyes the content and find what they were searching for, or it could just should a condensed view of the `archive` page listing all posts and pages.

### Hugo version

0.122.0

### Theme version

1.0.0

### What browsers are you seeing the problem on?

Chrome

### Link to Minimal Reproducible Example

https://optimizedbyotto.com/asdf

**Comment by mansoorbarri** [comment]
true, I was thinking the same. 

I used to use [GoKarna](https://github.com/526avijitgupta/gokarna) and they have "dynamic" 404 pages where the symbol changes everytime:

[screen-capture.webm](https://github.com/CaiJimmy/hugo-theme-stack/assets/90044118/1798dfc8-b012-40dd-8cb6-b528c2cfe07d)

this with search page would be pretty decent

**Comment by ottok** [comment]
The example from Gokarna is pretty, but it has no features that would help visitors find the content they were trying to see before they hit the 404 page.

A better example of functionality is e.g. https://seravo.com/something-that-does-not-exist which will show both search button and partial sitemap in the view to guide visitors forward:
![image](https://github.com/CaiJimmy/hugo-theme-stack/assets/668724/bc7e39bb-2e18-412b-897a-383fc0ee1a50)

**Comment by mansoorbarri** [comment]
true, it already shows the menu on the right. We can add the search & tags on the left side; make it look like the home page

**Comment by Himanshuch8055** [comment]
This issue is still open. Can I work on it? Please assign this issue to me.

**Comment by CaiJimmy** [comment]
> This issue is still open. Can I work on it? Please assign this issue to me.

Hi @Himanshuch8055, thank you for your interest in this issue! I have implemented a first version of this feature request, but I have not uploaded it to the repository yet. I'll upload it as soon as I can and we can see how we can improve it.

**Comment by CaiJimmy** [comment]
> > > This issue is still open. Can I work on it? Please assign this issue to me.
> > 
> > 
> > Hi @Himanshuch8055, thank you for your interest in this issue! I have implemented a first version of this feature request, but I have not uploaded it to the repository yet. I'll upload it as soon as I can and we can see how we can improve it.
> 
> No problem at all. Any time, definitely sure.

Sorry for the delay, I have just submitted it: #1081 

Live demo: https://dev.stack.jimmycai.com/p/math-type

**Comment by Himanshuch8055** [comment]
> > > > This issue is still open. Can I work on it? Please assign this issue to me.
> > > 
> > > 
> > > Hi @Himanshuch8055, thank you for your interest in this issue! I have implemented a first version of this feature request, but I have not uploaded it to the repository yet. I'll upload it as soon as I can and we can see how we can improve it.
> > 
> > No problem at all. Any time, definitely sure.
> 
> Sorry for the delay, I have just submitted it: #1081 
> 
> Live demo: https://dev.stack.jimmycai.com/p/math-type

No need to apologize! It's all good 😊. Thanks for your understanding.

**Comment by ottok** [comment]
This was closed as completed, but there is no automatic link to a git commit that closed it?

**Comment by ottok** [comment]
Seems decision to close this issue came from applying bugfix e8fc315ead9cb9a534355f9a860a392341f9524d. Other than that I don't see any changes since October 2024 in https://github.com/CaiJimmy/hugo-theme-stack/commits/master/layouts/404.html nor https://github.com/CaiJimmy/hugo-theme-stack/blob/master/assets/scss/partials/layout/404.scss

Testing the demo at https://demo.stack.jimmycai.com/this-is-a-demo I see it automatically searches on with the slug, which is good

**Comment by CaiJimmy** [comment]
Yes. I closed this issue because the search function and the automatic addition of the page URL to the search field on the 404 page had already been implemented. But I had forgotten to close the issue.
